### PR TITLE
Remove envy

### PR DESCRIPTION
**Pull request by peterjah**
NFT code as been changed here just to pass the as-pect compile step but should not be functionnal.
It will be changed to use args, and unit tests to use aspect, and moved in massa-sc-contract. but in another issue !

### CODE REVIEW

File: vm-mock/vm.js [hunk 1]
```diff
@@ -193,7 +193,7 @@ export function createMockedABI(memory, createImports, instantiate, binary) {
         return webModule.__newArrayBuffer(newAddress._value);
       },
       assembly_script_get_time(){
-        return BigInt(Date.now());
+        return BigInt.asUintN(64, Date.now());
```
Comment (Thykof) [inline vm-mock/vm.js:196]: not the good one

**Review by Thykof — CHANGES_REQUESTED** [review]

**Review by Thykof — APPROVED** [review]

**Review by Thykof — APPROVED** [review]

File: assembly/NFT/NFT.ts [hunk 2]
```diff
@@ -48,7 +48,7 @@ export function setNFT(_: string): string {
  * @param {string} newBaseURI new link include in the NFTs
  * @return {string}
  */
-export function setURI(_args: string): string {
+export function setURI(_args: StaticArray<u8>): string {
```
Comment (SebastienValla) [inline assembly/NFT/NFT.ts:51]: can we return string instead of StaticArray now ?

File: assembly/NFT/__tests__/NFT.spec.ts [hunk 3]
```diff
@@ -20,74 +20,66 @@ const transferAddress = new Address(
...
-    const want = 'massa.net/nft/1';
-    if (got != want) {
-      error(got.toString() + ', ' + want.toString() + ' was expected.');
-      return TestResult.Failure;
-    }
-    return TestResult.Success;
+    const tokenUri = tokenURI(args.serialize());
+    expect(tokenUri).toBe('massa.net/nft/1');
+
+    // if (got != want) {
```
Comment (SebastienValla) [inline assembly/NFT/__tests__/NFT.spec.ts:42]: commented code

File: assembly/NFT/__tests__/NFT.spec.ts [hunk 4]
```diff
@@ -20,74 +20,66 @@ const transferAddress = new Address(
...
+    //   error(got.toString() + ', ' + want.toString() + ' was expected.');
+    //   return TestResult.Failure;
+    // }
+    // return TestResult.Success;
   });
-  test('BaseURI call', (): i32 => {
+  xtest('BaseURI call', () => {
```
Comment (SebastienValla) [inline assembly/NFT/__tests__/NFT.spec.ts:48]: xtest

File: assembly/NFT/__tests__/NFT.spec.ts [hunk 5]
```diff
@@ -20,74 +20,66 @@ const transferAddress = new Address(
...
     const got = baseURI('');
     const want = 'massa.net/nft/';
-    if (got != want) {
-      error(got.toString() + ', ' + want.toString() + ' was expected.');
-      return TestResult.Failure;
-    }
-    return TestResult.Success;
+    expect(got).toBe(want);
+
+    // if (got != want) {
```
Comment (SebastienValla) [inline assembly/NFT/__tests__/NFT.spec.ts:53]: idem

**Review by SebastienValla — CHANGES_REQUESTED** [review]

**Review by SebastienValla — APPROVED** [review]

File: assembly/NFT/NFT.ts (hunk 2 above)
Comment (gregLibert) [inline assembly/NFT/NFT.ts:51]:
```suggestion
export function setURI(_args: StaticArray<u8>): StaticArray<u8> {
```

File: assembly/NFT/NFT.ts [hunk 6]
```diff
@@ -88,10 +88,9 @@ export function symbol(_: string): string {
  * @param {string} tokenId
  * @return {string}
  */
-export function tokenURI(_args: string): string {
+export function tokenURI(_args: StaticArray<u8>): string {
```
Comment (gregLibert) [inline assembly/NFT/NFT.ts:91]: Idem,and so on!

File: assembly/NFT/__tests__/NFT.spec.ts (hunk 5 above)
Comment (gregLibert) [inline assembly/NFT/__tests__/NFT.spec.ts:53]: dead code should be remove or uncommented

**Review by gregLibert — CHANGES_REQUESTED** [review]
If migrating properly the NFT is too long, you should diretcly remove its content from here and create an issue to add it to sc-contract repository.

**Comment by peterjah** [comment]
> If migrating properly the NFT is too long, you should diretcly remove its content from here and create an issue to add it to sc-contract repository.

ok! the issue is created and @SebastienValla is already working on it !
https://github.com/massalabs/massa-sc-contracts/issues/23

So I will remove it in this PR! I love remove code :smiley:

**Review by gregLibert — APPROVED** [review]
:clap: :camping:

**Pull request merged**